### PR TITLE
Hash device and module id before upload

### PR DESCRIPTION
**Pull request by lfitchett**
This is required to comply with GDPR. It replaces any device Ids with "device". Device Id's are replaced instead of hashed because compliance determined that deviceId hashes must use a rotating salt. 

Module id is simply hashed.

It also changes host module tag to module_name to match agent metrics

### CODE REVIEW

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Diagnostics/MetricTransformer.cs [hunk 1]
```diff
@@ -61,7 +70,7 @@ public MetricFilter AddAllowedTags(params KeyValuePair<string, string>[] pairs)
             return this;
         }
 
-        public MetricFilter AddTagsToAdd(params KeyValuePair<string, string>[] pairs)
+        public MetricTransformer AddTagsToAdd(params KeyValuePair<string, string>[] pairs)
         {
             if (this.tagsToAdd.HasValue)
```
Comment (and-rewsmith) [inline edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Diagnostics/MetricTransformer.cs]: nit: w1 could be more descriptive name

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Diagnostics/MetricTransformer.cs [hunk 2]
```diff
@@ -30,13 +31,21 @@ public IEnumerable<Metric> FilterMetrics(IEnumerable<Metric> metrics)
                     continue;
```
Comment (and-rewsmith) [inline edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Diagnostics/MetricTransformer.cs]: So allowed tags will bypass transformation? What is the reason for naming this "allowed"? What about "tagsIndicatingNoTransform"

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Diagnostics/MetricsWorker.cs [hunk 3]
```diff
@@ -125,6 +126,32 @@ async Task BeginUploadRetry(CancellationToken cancellationToken)
...
+        string ReplaceDeviceId(string id)
+        {
+            const string deviceIdReplacement = "device";
+
+            // Id is in the form of 'deviceId/moduleId'
+            string[] parts = id.Split('/');
+            if (parts.Length == 2)
+            {
+                parts[0] = deviceIdReplacement;
+                parts[1] = parts[1].CreateSha256(); // Hash moduleId
```
Comment (damonbarry) [inline edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Diagnostics/MetricsWorker.cs]: Any benefit to **NOT** hashing the module ID if it's a system module (i.e., begins with `$`)?

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Diagnostics/MetricTransformer.cs (hunk 2 above)
Comment (lfitchett) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Diagnostics/MetricTransformer.cs]: nope, this acts as a whitelist. Only metrics that contain the given tags will be allowed through, all others are dropped. It used to be called whitelist, but was renamed.

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Diagnostics/MetricsWorker.cs (hunk 3 above)
Comment (lfitchett) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Diagnostics/MetricsWorker.cs]: easier to read, I'll change it.

**Pull request merged**